lib/pos_mempool.go

```diff
@@ -721,61 +726,45 @@ func (mp *PosMempool) GetIterator() MempoolIterator {
 	return NewPosMempoolIterator(mp.txnRegister.GetFeeTimeIterator())
 }
 
-// Refresh can be used to evict stale transactions from the mempool. However, it is a bit expensive and should be used
-// sparingly. Upon being called, Refresh will create an in-memory temp PosMempool and populate it with transactions from
-// the main mempool. The temp mempool will have the most up-to-date readOnlyLatestBlockView, Height, and globalParams. Any
-// transaction that fails to add to the temp mempool will be removed from the main mempool.
-func (mp *PosMempool) Refresh() error {
-	mp.Lock()
-	defer mp.Unlock()
-
+// Any transaction that fails to add to the temp mempool will be removed from the main mempool.
+func (mp *PosMempool) validateTransactions() error {
+	mp.RLock()
 	if !mp.IsRunning() {
```

Comment:
Normally when you have more than one line in a Lock/RLock we like to break it up into a function like:

func getThings() {
    mp.RLock()
    defer mp.RLock()

    validationView := mp.readOnlyLatestBlockView
	mempoolTxns := mp.getTransactionsNoLock()

    return validationView, mempoolTxns
}

Reply:
This prevents disasters as people modify the code in the future.